src/main/java/com.company/linkedlist/DetectCycle.java

```diff
@@ -0,0 +1,27 @@
+package com.company.linkedlist;
+
+public class DetectCycle {
+    public boolean detectCycle(ListNode head) {
+        ListNode tortoise = head;
+        ListNode hare = head;
+
+        while (hare != null && tortoise != null) {
+            if (hare == null || hare.next == null) {
```

Comment:
This condition could be in the while loop. But not necessary.

Reply:
okay